doc/reference/randomness.rst

```diff
@@ -0,0 +1,86 @@
+.. _randomness:
+
+Randomness
+==========
+.. currentmodule:: networkx
+
+Random Number Generators (RNGs) are often used when generating, drawing
+and computing properties or manipulating networks. NetworkX provides
+functions which use one of two standard RNGs: NumPy's package `numpy.random`
+or Python's built-in package `random`. They each provide the same 
+algorithm for generating numbers (Mersenne Twister). Their interfaces
+are similar (dangerously similar) and yet distinct. 
+They each provide a global default instance of their generator that
+is shared by all programs in a single session.
+For the most part you can use the RNGs as NetworkX has them set up and
+you'll get reasonable pseudorandom results (results that are statistically
+random, but created in a deterministic manner). 
+
+Sometimes you want more control over how the numbers are generated.
+In particular, you need to set the `seed` of the generator to make
+your results reproducible -- either for scientific publication or 
+for debugging. Both RNG packages have easy functions to set the seed
+to any integer, thus determining the subsequent generated values.
+Since this package (and many others) use both RNGs you may need to
+set the `seed` of both RNGs.  Even if we strictly only used one of the
+RNGs, you may find yourself using another package that uses the other.
+Setting the state of the two global RNGs is as simple setting the
+seed of each RNG to an arbitrary integer:
+
+.. nbplot::
+
+   >>> import random
+   >>> random.seed(246)        # or any integer
+   >>> import numpy
+   >>> numpy.random.seed(4812)
+
+Many users will be satisfied with this level of control.
+
+For people who want even more control, we include an optional argument
+to functions that use an RNG.  This argument is called `seed`, but
+determines more than the seed of the RNG. It tells the function which
+RNG package to use, and whether to use a global or local RNG.
+
+.. nbplot::
+
+    >>> seed=None    # use the global default RNG of either package
```

Comment:
If I'm understanding this correctly, these are examples of how I might set the `seed` keyword argument in an invocation of a NetworkX function. It might be clearer to show these examples as used in an actual function call. For example,

```python
>>> random_graph(n, m, p, seed=None)
>>> random_graph(n, m, p, seed=42)
# etc.
```

Reply:
Good -- thanks!  I agree...